src/katello/tracer/dnf.py

```diff
@@ -0,0 +1,66 @@
+import re
+import os
+import subprocess
+from katello.tracer.SystemdDbus import SystemdDbus
+
+class DnfTracerApp:
+    def __init__(self, name, helper, type):
+        self.name = name
+        self.helper = helper
+        self.type = type
+
+
+def get_name(pid):
+    bus = SystemdDbus()
+    if pid and bus.unit_path_from_pid(pid):
+        if not bus.has_service_property_from_pid(pid, 'PAMName'):
+            Id = bus.get_unit_property_from_pid(pid, 'Id')
+            if Id and re.search("\.service$", Id):
+                return re.sub('\.service$', '', Id)
+    return None
+
+
+def run_command(command):
+    os.environ['LANG'] = 'C'
+    process = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+   
+    return_output = ''
+    return_code = process.wait()
+    
+    if return_code == 0:
+        output = process.stdout.read().decode('utf-8')
+        return_output = output.strip()
+    return return_code, return_output
+
+
+def collect_services_state():
+    command = "dnf needs-restarting"
+    retval, output = run_command(command)
+
+    lines = output.split('\n')
```

Comment:
Going with the suggestion from above to avoid `run_command`, it'd be:
```suggestion
    env = dict(os.environ, LANG='C')
    process = subprocess.run(['dnf', 'needs-restarting'], env=env, stderr=subprocess.STDOUT, text=True)

    if process.returncode != 0:
        return []

    lines = process.stdout.split('\n')